app/registries/foreman/plugin.rb

```diff
@@ -315,6 +316,14 @@ def add_all_permissions_to_default_roles
       end
     end
 
+    def fixture_directory(directory = nil)
+      if directory
+        @fixture_directory = directory
+      else
+        @fixture_directory
+      end
```

Comment:
This is clunky, what is the idea here? Is this reader or writer? Reader was already defined above.

Frankly, I think the most flexible API would be `attr_reader fixtures` and `add_fixtures(*files)` methods so plugins could call `add_fixtures(Dir["#{plugin_root}/test/{**,*}/*.{yml}"].uniq)` or something like that.

Reply:
I was attempting to mimick the plugin api, which seems to rely on the attribute and the 'setter' being named differently, which is clunky in and of itself :)

Reply:
Having the plugin specify the files seems clunky to me TBH, when they're all going to be in a directory

Reply:
Indeed an assumption we can make, or not. I am slightly towards exposing it on file-level, but I can live with directory. I'd perhaps do one line if in this case.

Anyway, our plugin API is kind of DSL, so this is the reason why we don't use attributes I think.